ToyBox/classes/MonkeyPatchin/BagOfPatches/MiscWrath.cs

```diff
@@ -675,6 +677,43 @@ private static void Postfix(Polymorph __instance) {
                 __instance.Owner.View.transform.localScale = new Vector3(scale, scale, scale);
             }
         }
+        [HarmonyPatch(typeof(LocalizedString))]
+        internal class BPTagger 
+        {
+            private static LocalizationPack pack = Kingmaker.Localization.LocalizationManager.LoadPack(Kingmaker.Localization.LocalizationManager.s_CurrentLocale);
+
+            [HarmonyPatch(nameof(LocalizedString.LoadString)), HarmonyPostfix]
+            public static void LoadString_ModTagPatch(ref string __result, LocalizedString __instance) 
+            {
+                if (settings.togglemoddedbptag) 
+                {
+                    try 
+                    {
+                        if (!__result.Contains(".") && !__result.Contains("。")) { return; }
+                        string actualKey = __instance.GetActualKey();
+                        if (!pack.m_Strings.TryGetValue(actualKey, out var _) ) 
+                        {
+                            __result += " (modded blueprint)";
+                        }
+                    } 
+                    catch (Exception e) 
+                    { 
+                        Main.logger.Log("Error patching LoadString for modded blueprint tagging  \n" + e);
+                    }
+                }
+            }
+        }
+
+        [HarmonyPatch(typeof(Kingmaker.Localization.LocalizationManager))]
+        internal class BPTagger_Pack 
+        {
+            [HarmonyPriority(Priority.First)]
+            [HarmonyPatch(nameof (Kingmaker.Localization.LocalizationManager.LoadPack)), HarmonyPostfix]
+            public static LocalizationPack LoadPack_ModTagPatch(LocalizationPack __result) 
+            {
+                return AccessTools.MakeDeepCopy<LocalizationPack>(__result);
```

Comment:
You're returning a value on a Postfix?
I think you meant to assign the DeepCopy to `BPTagger.pack` instead of returning it...

Reply:
:owlcat_laugh: